src/python/pants/reporting/reporting.py

```diff
@@ -95,14 +97,20 @@ def initial_reporting(self, run_tracker):
   def _get_invalidation_report(self):
     return InvalidationReport() if self.get_options().invalidation_report else None
 
+  @staticmethod
+  def _consume_stringio(f):
+    f.flush()
+    buffered_output = f.getvalue()
+    f.close()
+    return buffered_output
+
```

Comment:
consider:

```
from contextlib import closing

@staticmethod
def _consume_stringio(f):
  with closing(f):
    f.flush()
    return f.getvalue()
```

Reply:
I don't think this has much value unless the creation of f is part of the idiom. Here f is created far away, and there already so many other opportunities for it to leak that I suspect that this will be more misleading than helpful.

Reply:
its really just a way to eliminate the `buffered_output` variable in this case by being able to directly return the output of `f.getvalue()`.